src/connector/src/sink/snowflake.rs

```diff
@@ -236,32 +228,105 @@ impl SnowflakeSinkWriter {
                 TimestamptzHandlingMode::UtcString,
                 TimeHandlingMode::String,
             ),
-            row_counter: 0,
-            payload: String::new(),
-            max_batch_row_num,
-            // initial value of `epoch` will start from 0
+            // initial value of `epoch` will be set to 0
             epoch: 0,
-        }
+            // will be (lazily) initialized after the begin of each epoch
+            // when some data is ready to be upload
+            streaming_uploader: None,
+        })
+    }
+
+    /// update the streaming uploader as well as the file suffix.
+    /// note: should *only* be called iff after a new epoch begins,
+    /// and `streaming_upload` being called the first time.
+    /// i.e., lazily initialization of the internal `streaming_uploader`.
+    async fn new_streaming_uploader(&mut self) -> Result<()> {
+        let file_suffix = self.file_suffix();
+        let path = generate_s3_file_name(self.s3_client.s3_path(), &file_suffix);
+        let uploader = self
+            .s3_client
+            .opendal_s3_engine
+            .streaming_upload(&path)
+            .await
+            .map_err(|err| {
+                SinkError::Snowflake(format!(
+                    "failed to create the streaming uploader of opendal s3 engine for epoch {}, error: {}",
+                    self.epoch,
+                    err
+                ))
+            })?;
+        self.streaming_uploader = Some((uploader, file_suffix));
+        Ok(())
+    }
+
+    /// should be *only* called after `commit` per epoch.
+    fn reset_streaming_uploader(&mut self) {
```

Comment:
This method seems unnecessary. We must have taken the streaming uploader out in `finish_streaming_upload`.